src/Analysis/Ast/Impl/Specializations/Typing/Types/GenericTypeParameter.cs

```diff
@@ -35,31 +35,48 @@ public GenericTypeParameter(string name, IPythonModule declaringModule, IReadOnl
 
 
         public static IPythonType FromTypeVar(IArgumentSet argSet, IPythonModule declaringModule, IndexSpan location = default) {
-            var args = argSet.Values<IMember>();
+            var args = argSet.Arguments;
+            var constraintArgs = argSet.ListArgument?.Values;
+
             if (args.Count == 0) {
                 // TODO: report that at least one argument is required.
                 return declaringModule.Interpreter.UnknownType;
             }
 
-            var name = (args[0] as IPythonConstant)?.GetString();
+            var name = (args[0].Value as IPythonConstant)?.GetString();
             if (string.IsNullOrEmpty(name)) {
                 // TODO: report that type name is not a string.
                 return declaringModule.Interpreter.UnknownType;
             }
 
-            var constraints = args.Skip(1).Select(a => {
+            var constraints = constraintArgs?.Select(a => {
```

Comment:
Not sure about this. `x?.Select` will yield null and `.ToArray` may then crash. Also, `constraints` is assumed to never be null below.

Reply:
ok